common/models/user.js

```diff
@@ -247,6 +247,12 @@ module.exports = function(User) {
             debug('An error is reported from User.hasPassword: %j', err);
             fn(defaultError);
           } else if (isMatch) {
+            if ((credentials.password).length > MAX_PASSWORD_LENGTH) {
```

Comment:
Actually, I take that back. Let's keep the validation here, to keep the response time roughly the same regardless of whether the password is less or more than MAX_PASSWORD_LENGTH long.

However, please add a comment explaining why it's important to keep the check here. For example:

```js
// Only check password length after we run the same code we would execute
// for a password with a valid length. This should prevent attackers from guessing
// the maximum allowed password length by measuring the response time.
```